packages/nimble-components/CONTRIBUTING.md

```diff
@@ -77,15 +77,33 @@ Before building a new component, 3 specification documents need to be created:
 
 ## Develop new components
 
-### Marking a component as in development
+### Marking a component as incubating
 
-If a component will require multiple pull requests before having a complete and stable API, it should be marked as "in-development" to indicate to clients that they shouldn't start using it yet. To do this:
+If a component is not ready for general use, it should be marked as "incubating" to indicate that status to clients. A component could be in this state if any of the following are true:
+
+-   It is still in development.
+-   It is currently experimental or application-specific and hasn't yet been generalized for broader use.
+-   It is missing important features like interaction design, visual design, accessibility, or framework integration.
+
+Incubating contributions may compromise on the above capabilities but they still must abide by other repository requirements. For example:
+
+-   Coding conventions (element naming, linting, code quality)
+-   Unit and Chromatic test coverage
+-   Storybook documentation
+
+To mark a component as incubating:
 
 1. In the component status table, set its status to ⚠️
-2. In the component Storybook documentation, add a red text banner to the page indicating that the component should not be used
-3. Consider placing the component implementation in a sub-folder named `experimental` so that it will be obvious when importing it that it is incomplete
+2. In the component Storybook documentation:
+    - add a red text banner to the page indicating that the component is not ready for general use
```

Comment:
Would be handy if the storybook doc page discussed what work was remaining to [graduate](https://incubator.apache.org/guides/graduation.html) from incubating. Like a checklist showing what is done and what is missing. Also gives a status of how mature an incubating component is.

Maybe we can capture in this section what that checklist includes but could be in a follow-up PR once we move some components to incubating and figure out the checklist.

Some items could be (kinda a rehash of the spec template):

- [ ] Follows coding conventions
- [ ] Unit test coverage
- [ ] Storybook matrix test coverage
- [ ] Storybook documentation
- [ ] Interaction design review
- [ ] Visual design review
- [ ] Accessibility implemented
- [ ] Angular integration
- [ ] Blazor integration

Reply:
I like the idea of promoting this information. But to avoid duplication, I propose we keep that checklist in the component issue (as we have for #924 and #309) but update the storybook incubating header to link to it. I'll plan to do this in the follow-up.